altinvoice.php

```diff
@@ -55,12 +55,17 @@ function altinvoice_civicrm_alterMailParams(&$params, $context) {
         'id' => $params['contactId'],
         'cs' => $checksum,
       ];
-      $payUrl = CRM_Utils_System::url('civicrm/user', $urlParams, TRUE);
-      $params['text'] .= "\nClick here to pay this invoice: $payUrl";
-      $params['html'] .= "<p>Click here to <a href='$payUrl'>pay this invoice</a>.</p>";
+
+      $isLinkFlagSet = Civi::settings()->get('altinvoice_include_link_to_pay');
+      if ($isLinkFlagSet) {
```

Comment:
These two lines should move further up above the "hack in a link to the invoice ID" comment - or even combined with the `if` statement immediately following that comment, and the comment updated to reflect the fact that it's conditional.

Reply:
I hope I didn't misunderstand this one!